modules/consensus/synchronize.go

```diff
@@ -175,39 +176,82 @@ func (cs *ConsensusSet) managedReceiveBlocks(conn modules.PeerConn) (returnErr e
 
 	// Read blocks off of the wire and add them to the consensus set until
 	// there are no more blocks available.
-	moreAvailable := true
-	for moreAvailable {
-		// Read a slice of blocks from the wire.
-		var newBlocks []types.Block
-		if err := encoding.ReadObject(conn, &newBlocks, uint64(MaxCatchUpBlocks)*types.BlockSizeLimit); err != nil {
-			return err
-		}
-		if err := encoding.ReadObject(conn, &moreAvailable, 1); err != nil {
-			return err
-		}
+	blocksChan := make(chan []types.Block, 1000)
```

Comment:
Max block size is 2mb, queuing 1000 blocks is not good. I think at best we could queue 100, but even that makes me uncomfortable. If you want to get fancier, we could track the size and then limit it to 50mb or somethin. And we could also add a settings endpoint to the consensus api to make the 50mb number configurable.